sweep/txgenerator.go

```diff
@@ -317,6 +264,7 @@ func getWeightEstimate(inputs []input.Input) ([]input.Input, int64, int, int) {
 	// Our sweep transaction will pay to a single segwit p2wkh address,
 	// ensure it contributes to our weight estimate.
 	weightEstimate.AddP2WKHOutput()
+	txWeight := int64(weightEstimate.Weight())
```

Comment:
The inputs and outputs are now all calculating their weights in isolation. I think things are being counted multiple times now, see implementation of `Weight()`. For example the BaseTxSize.

Reply:
Very good catch! I did not expect the `WeightEstimator` to behave this way. Fixed this and also added a test to lock down the fee estimation.
Ran the test against master and got the same results (apart from four witness types that didn't have a size implemented in master before the refactor).